### PR TITLE
fix(link-list-section): Center align link list items in multi column layouts

### LINKED ISSUE #6830: Link List item not using same height
## Detailed description

> Describe in detail the issue you're having.
We use a different web component for the link (dds-link-list-item-cta) but same problem is with dds-link-list-item. I found that if i chnge the a-tag inside the linklist item to flex !important it works. But on storybook it works without. No idea why the flex is not working for us no matter if we use the same link list item as storybook or our.

> Is this a feature request (new component, new icon), a bug, or a general
> issue?
Bug

> Is this issue related to a specific component?
Link List

> What did you expect to happen? What happened instead? What would you like to
> see changed?
Items should use same height and center text

> What browser are you working in?
Chrome, Safari, Firefox

> What version of Carbon for IBM.com are you using?
1.21

> What offering/product do you work on? Any pressing ship or release dates we
> should be aware of?
IBM CMS

## Steps to reproduce the issue

1. add link list
2. In one item add as much text as needed to create a second line

## Additional information
As said this works in storybook but not for us even with the web component version used in sotrybook

Screenshot 2021-08-06 at 13 40 07

Because this bug are visible for everyone i can not post a link to the page with the problem. But please contact me over Slack (Christian Schimanski) if you have questions or need the link to the sample page

----
HC Jira issue:  (https://jsw.ibm.com/browse/HC-2139) 
HC Dev:   Andy Blum
Story Points:  3
DDS consultant:  Ignacio Becerra

### PR DESCRIPTION
**Pull request by andy-blum**
### Related Ticket(s)

Fixes #6830 

### Description

- Sets `link-list-item` elements within `link-list__list--split` and `link-list__list--three-columns` to `display:grid`
- Updates `link-list-section` storybook config to have one item with a very long title.

Screen Shot 2021-08-17 at 4 51 14 PM

### Changelog

**Changed**

- web component link-list items in 2 & 3 column layouts display as grid & center-align their contents
- web component link-list-section has one item with very long title text

### CODE REVIEW

**Comment by IgnacioBecerra** [comment]
@andy-blum Thanks for opening this PR, looks great with the fix! 

Looks like you just have to run `yarn format` so the ci-check can pass 👍

**Review by shixiedesign — CHANGES_REQUESTED** [review]
Hi! Thanks for opening a fix, though this is not the correct design. The expected result is still top aligned link texts. The current bug (please [see Anna's comment to clarify](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues/6830#issuecomment-897640939)) is with hover's background color change. Both icon and link text should be 16px from the top rule, not centered: 

image

image

**Comment by shixiedesign** [comment]
A little more detailed spec: 
image

**Comment by shixiedesign** [comment]
@andy-blum Whenever you are ready for another look, please remove `⚒️ fix needed` label, and add the label `👀 eyes needed`. Thanks!

**Comment by shixiedesign** [comment]
Thank you! @andy-blum Much better and really close - only 1 comment, the spacing between text and arrow icon should be 32px. 

**Currently 8px:**
image

**Expecting 32px** 
image

**Comment by andy-blum** [comment]
@shixiedesign I just updated the icon's padding value, but I'm hesitant to update the styles for the link-list default view as the center-alignment there comes from dds-card-footer

Screen Shot 2021-08-19 at 2 47 00 PM

**Comment by shixiedesign** [comment]
@andy-blum Hmm thanks for checking! I just looked at our Cards, and seems like the link text simply doesn't wrap 😱 I just filed [a bug issue for this](https://app.zenhub.com/workspaces/carbon-for-ibmcom-5d449f3642eb1962336cbe52/issues/carbon-design-system/carbon-for-ibm-dotcom/6935). But I'm gonna check with our dev to see what's the usual approach when this happens. More soon.

**Comment by kennylam** [comment]
@shixiedesign This icon alignment should be consistent for `card` and all components which use it, correct? If so then changing `dds-card-footer` should be fine @andy-blum.

**Comment by andy-blum** [comment]
@shixiedesign pinging you since I can't update labels - this should be ready for your re-review

**Comment by shixiedesign** [comment]
Almost there! Only noticed in Link list - default, at `sm` breakpoint, missing 32px spacer between arrow and wrapping text 
image

**Comment by andy-blum** [comment]
@shixiedesign 👀 eyes needed

**Comment by shixiedesign** [comment]
@andy-blum Hey looks like you didn't contain the change to only mobile? Desktop of Link list - default looks off now: 
image

**Review by shixiedesign — APPROVED** [review]
Perfect. Thank you! 👍

**Comment by andy-blum** [comment]
Forgot to run `yarn format` for CI checks.

**Comment by shixiedesign** [comment]
Hey @andy-blum if you still have a second, can we shorten the link text in Link list section to `Learn more about Kubernetes and automating deployment `. It's still gonna wrap so good for future testing. Thanks!

File: packages/web-components/src/components/link-list-section/__stories__/link-list-section.stories.ts [hunk 1]
```diff
@@ -20,7 +20,8 @@ export const Default = ({ parameters }) => {
       <dds-link-list-heading>${heading}</dds-link-list-heading>
       <dds-link-list>
         <dds-link-list-item href="https://example.com">
-          Learn more about Kubernetes ${ArrowRight20({ slot: 'icon' })}
+          Learn more about Kubernetes, an open-source system for automating deployment, scaling, and management of containerized
+          applications. ${ArrowRight20({ slot: 'icon' })}
```
Comment (emyarod) [inline packages/web-components/src/components/link-list-section/__stories__/link-list-section.stories.ts:24]: I think we can shorten this to `Learn more about Kubernetes and automating deployment`

**Review by emyarod — COMMENTED** [review]
looks good to me, just one note about the storybook example

**Comment by jeffchew** [comment]
Hi @andy-blum ! Can you check on the following stylelint issue?

image

**Review by jeffchew — APPROVED** [review]
LGTM!
